protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -746,24 +747,28 @@ func (rpccs *RPCConsumerServer) GetInitialUnwantedProviders(directiveHeaders map
 	return unwantedProviders
 }
 
-func (rpccs *RPCConsumerServer) HandleDirectiveHeadersForMessage(chainMessage chainlib.ChainMessage, directiveHeaders map[string]string) {
-	timeoutStr, ok := directiveHeaders[common.RELAY_TIMEOUT_HEADER_NAME]
-	if ok {
-		timeout, err := time.ParseDuration(timeoutStr)
-		if err == nil {
-			// set an override timeout
-			chainMessage.TimeoutOverride(timeout)
-		}
-	}
+func (rpccs *RPCConsumerServer) getExtensionsFromDirectiveHeaders(directiveHeaders map[string]string) []string {
```

Comment:
best if this function gets the latest block, directive headers and returns ExtensionInfo